### ISSUE TITLE
Equilibrium.get_flux_surface interferes with user's matplotlib code

### ISSUE THREAD

**Issue by LiamPattinson** [bpatel2107/pyrokinetics#60]
```python
from pyrokinetics.equilibrium import Equilibrium
import matplotlib.pyplot as plt
import numpy as np

# Do some plotting
x = np.linspace( 0, 2*np.pi, 1000)
y = np.sin(x)
plt.plot(x,y)

# get flux surfaces
eq = Equilibrium("pyrokinetics/templates/transp_eq.cdf", "TRANSP")
R_con, Z_con = eq.get_flux_surface(1.0)

plt.show() # nothing happens!
```
When using `get_flux_surface`, it deletes any work the user may have been doing with matplotlib. The cause lies with the calls to `mpl.use`:

https://github.com/pyro-kinetics/pyrokinetics/blob/766ec1afa1875a3f1923b6404180ddf8eddcea78/pyrokinetics/equilibrium.py#L383-L388
https://github.com/pyro-kinetics/pyrokinetics/blob/766ec1afa1875a3f1923b6404180ddf8eddcea78/pyrokinetics/equilibrium.py#L425

If you comment out the backend changing lines, and replace the contour line with this:

```python
fig, ax = plt.subplots()
con = ax.contour(self.R, self.Z, psin_2d, levels=[psi_n])
plt.close(fig)
```
everything runs just fine, but the tests now take 3 times as long to run. Ideally, there should be a way to retrieve the contour without matplotlib. Perhaps [skimage](https://scikit-image.org/docs/dev/auto_examples/edges/plot_contours.html) would help? It uses the same underlying algorithm, but I'd need to do some side-by-side testing to ensure it can deliver the same results.

**Comment by bpatel2107** [comment]
I had a go with skimage but it often struggled to find the flux surfaces as one continuous contour. Rather it found multiple contours that look linked by eye but aren't in the code. I believe I used the following method

https://scikit-image.org/docs/dev/api/skimage.measure.html#skimage.measure.find_contours

Maybe there's a better way or I didn't do a good job the first time.

The reason I did the backend switch was for speed as you see, but clearing the figure like this is not great. The only other option I can think of is to save any existing figures before changing the backend though I don't know if that's fast/better
